src/agents/tests/matrix.sh

```
@@ -0,0 +1,350 @@
+#!/usr/bin/env bash
+# Agents tool — comprehensive matrix test
+# Runs all permutations and asserts exact behavior.
+
+set -uo pipefail
+cd /Users/Martin/Tresors/Projects/GenesisTools
```

Shellcheck (0.11.0)

[warning] Use 'cd ... || exit' or 'cd ... || return' in case cd fails.

(SC2164)